react/components/Provider/Warehouse.tsx

```diff
@@ -154,6 +154,12 @@ const WarehouseProvider: FC = (props) => {
     setDate(undefined)
     setKeep(undefined)
     setItems(undefined)
+
+    if (dataListSupplyLots?.listSupplyLots?.length > 9) {
+      if (!limit) setLimit(true)
+    } else if (limit) {
+      setLimit(false)
+    }
```

Comment:
Sugestão de alteração para melhorar a legibilidade da condição, se achar quer faz sentido:

```suggestion
    if (dataListSupplyLots?.listSupplyLots?.length > 9 && !limit) {
      setLimit(true)
    } else if (limit) {
      setLimit(false)
    }
```

Reply:
Eu tinha feito essa condição assim antes, mas acontece que dependendo do caso ela não funciona como deveria, então fiz essa alteração que passou a funcionar corretamente.
Se quiser posso te explicar melhor no slack!